### DISCUSSION TITLE
Multi-Contrast Multi-Sessions Feature Potential Discussions.

### DISCUSSION THREAD

**Discussion by dyt811** [ivadomed/ivadomed#1003] [Ideas]
Hi all. 

We are finally at a sufficiently refactored and base code improved stage to merge earlier radical experimental branch of changes. This would be a good time to revisit some key assumptions which are needed to resolve challenges with adopting multi-sessions/contrasts support: 

## Key Assumptions: 

* All modality/session input data are coregistered by user already (and of same dimension!). 

* For a subject to be included, it must have ALL [`Targeted Modality`] x [`Targeted Session`] data. Missing any session or modality will result in exclusion. 
   * Rationale: if we start accommodating either missing modalities or missing sessions, the input vector will not be stable (since they need to be FIXED dimension) and will lead to require some very non-scalable solution to guess which session/modality to use for the missing ones or interpolation across modality/session which require many experiment specific knowledge. 
   * For the initial release, we focus on the bare minimal cases: all modality all session data are present, fixed and of **same dimension** at least within each subject. 
  
## Clarification Questions: 

* Currently, it seems we don't require all `target_suffix` data to be present (I believe?). Keeping this? 
   * Rationale: if all are data are coregistered with each other, at least ONE of the `target_suffix` (aka ground truth) is sufficient?  

* What about `roi_suffix`? Right now it is incorporated into `target_suffix` and the code just check if it is one of the derivatives detected? 

* Should we consider ground truth outside of the targeted modality? See below: Currently STRICT [target modality] match.
![image](https://user-images.githubusercontent.com/6720068/145754667-52378b70-9c00-48b3-b804-27fe02890947.png)

**Comment by jcohenadad** [comment]
`roi_suffix`: Name of the variable that is used via the config file field [roi_params](https://ivadomed.org/configuration_file.html?highlight=roi_params#roi-params). It points to a mask that we can use as an ROI, used for cropping.
`target_suffix`: Designates the ground truth. See [documentation](https://ivadomed.org/configuration_file.html?highlight=target_suffix#target-suffix)

**Comment by dyt811** [comment]
Per answers gathered from the meeting: 

1) We want to keep this. As long as at least one ground truth is defined. We keep the subject. 
   * Example: if subject has t1 target_suffix when config parameters requires both t1 and t2 target suffix (a list of t1/t2). 

2) ROI suffix is used to define ROI region (as a kind of spatial filtering). Combining ROI suffix and Target suffix for indexing purpose is FINE BUT we cannot leave ROI suffix inside target suffix for ground truth definition. 
   * [The `suffix` property in `roi_params`](https://ivadomed.org/configuration_file.html?highlight=roi_params#roi-params) is not required for all studies 
   * `target_suffix` are essentially ground-truth, see [Documentation](https://ivadomed.org/configuration_file.html?highlight=target_suffix#target-suffix)
   * The `suffix` property in `roi_params` is not related to ground truth (for training purposes). It helps refines the ground truth to reduce region used for training. 
   * In the current code, `roi_suffix` is added to `target_suffix` for file indexation in bids_dataframe, but it **could** be kept separated if that is an issue


3) Yes, we should consider additional ground truth defined in target modality. 
   
   * ![image](https://user-images.githubusercontent.com/6720068/145876621-f1dd6997-610d-4c1e-85e5-9c69a28b3e5e.png) 
    Regarding the screenshot above: i.e. we will use the red bound boxes data. T2star ground truth. 
   * eg: modality T1w and T2w, if a GT only exists in FLAIR, we keep it).

**Comment by dyt811** [comment]
Moved to issue to continue the internal discussion.  #1009 

Closing it here.
